src/ffi_sse2.rs

```diff
@@ -73,6 +73,10 @@ pub unsafe fn hash_many<const N: usize>(
 }
 
 pub mod ffi {
+    #[cfg_attr(
+        blake3_sse2_ffi,
+        link(name = "blake3_sse2_sse41_avx2_assembly", kind = "static")
```

Comment:
All of these extensions have assembly implementations and intrinsics implementations, which we switch between with `--feature prefer_intrinsics`. Do they all need the same treatment as AVX-512?

More broadly, could you help me understand what the `link(...)` attribute actually means :)

Reply:
I started digging into this and actually now I have questions about why the compiler seems to correctly forward native libraries dependencies through the crate graph if they come from `#[link(` but not if they come from command line arguments. I've poked some other compiler people, and I'll try to get back to you soon.